### PR TITLE
TOOLS-2557 Add rust tooling makefiles

### PR DESCRIPTION
**Pull request by travispaul**
This change provides eng consumers with `tools/mk/Makefile.rust.defs` and `tools/mk/Makefile.rust.targ` to use in their Makefiles.

At a minimum, the Makefile needs to set `RUST_TOOLCHAIN` to the desired Rust toolchain version. On Illumos systems, by default, the toolchain will be downloaded from the [Bootstrap kits for SmartOS/illumos](https://us-central.manta.mnx.io/pkgsrc/public/pkg-bootstraps/index.html). To instead use `rustup` the `RUST_USE_BOOTSTRAP` can be set to `false`.

Additional testing notes are provided on [TOOLS-2557](https://mnx.atlassian.net/browse/TOOLS-2557) and the existing consumers of eng have been tested against these changes (please let me know if I missed any):

- https://github.com/TritonDataCenter/firewall-logger-agent/pull/23 
- https://github.com/TritonDataCenter/manta-buckets-mdapi/pull/63 
- https://github.com/TritonDataCenter/rust-illumos-priv/pull/8 
- https://github.com/TritonDataCenter/manta-rebalancer/pull/195 
- https://github.com/TritonDataCenter/sastopo2svg/pull/2

If this PR is approved I'll update the branches in the above repos and make the PRs as "Ready to review."

[TOOLS-2557]: https://mnx.atlassian.net/browse/TOOLS-2557?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by danmcd — APPROVED** [review]
Thank you for the testing notes.  I'm most worried about regressions, and with the Triton release coming up in two weeks, it seems you won't be regressing anything in there with this.

You *probably* should have another reviewer with more Rust-savvy, but I'm happy approving this given how many repos you tested.

Also, will you have to update `deps/eng` / `.gitmodules` for these repos-under-test?

**Comment by travispaul** [comment]
> I'm most worried about regressions

The good news here is they they're building with the same version of Rust as before it's just not from pkgsrc (though they are pkgsrc bootstrap builds so should be pretty close to identical.)

> Also, will you have to update `deps/eng` / `.gitmodules` for these repos-under-test?

yes, as of right now they all have a PR open pointing to this branch once it's merged I'll update the PRs to master eng branch.

**Review by bahamat — APPROVED** [review]
I think this approach looks good.

**Pull request merged**
